pyvista/core/datasetattributes.py

```diff
@@ -750,8 +749,23 @@ def _prepare_array(
         """Prepare an array to be added to this dataset."""
         if data is None:
             raise TypeError('``data`` cannot be None.')
-        if isinstance(data, Iterable):
-            data = pyvista_ndarray(data)
+
+        # attempt to reuse the existing pointer to underlying VTK data
+        if isinstance(data, pyvista_ndarray):
+            # pyvista_ndarray already contains the reference to the vtk object
+            # pyvista needs to use the copy of this object rather than wrapping
+            # the array (which leaves a C++ pointer uncollected.
+            if data.VTKObject is not None:
+                # VTK doesn't support strides, therefore we can't directly
+                # point to the underlying object
+                if data.__array_interface__['strides'] is None:
```

Comment:
Is this the same as a check for being c-contiguous? If so, a dunder-free alternative:
```suggestion
                if data.flags.c_contiguous:
```